kep/108-Memory-bandwidth-and-LLC-aware-filtering-and-scoring/README.md

```diff
@@ -0,0 +1,145 @@
+    * This means that for a pod, all the nodes will be sorted by memory bandwidth, memory latency, cpu, LLC contention and LLC utilization
+    * All these lists will then go on to the next step
+
+
+* the number of counterweights that are applicable to the node
+    * A counterweight is a pod that has already been deployed on a node, but we assume it is still too soon to notice its impact in the measurements we receive from Prometheus. Therefore, we add its profile (memory bandwidth, cpu, and LLC utilization for now) as a counterweight for a *COUNTERWEIGTH_EXPIRATION_POLICY* period of time to the respective node
+
+* ranking strategy when giving point to the sorted nodes
+    * We allow flexible point awarding for our scheduler plugin. For example, our default strategy is to award 10 points to the node ranking first in a certain resource, 5 points to the second place, 1 point to the 3rd place and 0 points to anyone else.
+
+* the pod affinity to the respective resource
+    * The pod affinity is also obtained via a top-down analysis. It can have values between 0 (no impact) and 100 (very heavy impact) of a resource upon the pod execution time. The affinity of a pod acts a multiplier for the points awared in the ranking algorithm.
+       Therefore, a node with a lot of available memory bandwith (ranking first) might not get a pod that needs less LLC thrashing in fact.
+
+For pod that are not profiled yet, we allocate a default profile.
+
+Therefore the overall score of a node is calculated by:
+
+     # Assuming our scoring method for a 
+     scoring_list = [10,5,1, [0]*n]
+     WEIGHT_MEMORY_BANDWIDTH = 10
+     WEIGHT_MEMORY_LATENCY = 3
+     WEIGHT_LLC_OCCUPANCY = 6
+     WEIGHT_LLC_MPKI= 2
+     WEIGHT_CPU = 5
+
+     node_score = scoring_list[position_of_node_ordered_by_memory_bandwidth] * WEIGHT_MEMORY_BANDWIDTH + 
+                  scoring_list[position_of_node_ordered_by_memory_latency] *  WEIGHT_MEMORY_LATENCY +
+                  scoring_list[position_of_node_ordered_by_llc_occupancy] *  WEIGHT_LLC_OCCUPANCY +
+                  scoring_list[position_of_node_ordered_by_llc_mpki] *  WEIGHT_LLC_MPKI +
+                  scoring_list[position_of_node_ordered_by_cpu] *  WEIGHT_CPU
+
+
+
+### Extension points
+
+#### Filter
+When filter a node the memory and memory bandwidth requirement are estimated accroding to the classification of the pod to be scheduled. we check the free memory and free memory bandwidth of each node. In order to avoid unnecessary oversubscription, an OVERPROVISIONING factor is ued when calculate the memory and memory bandwidth to be reserved. This factor is got from the config file of this plugin, by default set to 2.
+
+#### PreScore
+At this extension point the name of the nodes to be scored are kept. At the Score extension point we will need rank these nodes to calculate the score.
+
+#### Score
+Calculate the score of a node according to the algorithm describe above.
```

Comment:
It'd be nice to give a concrect example to showcase the full story.

Reply:
done